client/src/pages/contentPages/Groups.js

```diff
@@ -1,41 +1,326 @@
+      textTransform: 'none',
+      height: '3rem',
+  },
+    paper: {
+      position: 'fixed',
+      width: 475,
+      backgroundColor: theme.palette.background.paper,
+      boxShadow: theme.shadows[5],
+      padding: theme.spacing(2, 4, 3),
+      borderRadius: 4,
+      outline: 'none'
+
+    },
+
+    groupImageContainer: {
+      flex: 1,
+      display: 'flex',
+      flexDirection: 'column',
+      justifyContent: 'center',
+      alignItems: 'center',
+      paddingTop: "10px",
+      paddingBottom: "10px"
+    },
+    large: {
+      height: "300px",
+      width: "400px",
+    },
+    uploading: {
+      opacity: 0.5,
+      height: "300px",
+      width: "400px"
     },
-}));
+    
+  }));
+
+
+  function getModalStyle() {
+    const top = 50;
+    const left = 50;
+  
+    return {
+      top: `${top}%`,
+      left: `${left}%`,
+      transform: `translate(-${top}%, -${left}%)`,
+    };
+  }
 
 const Groups = () => {
-    const classes = useStyles();
+
+  const classes = useStyles();
+  const { userCourse, profile } = useGlobalContext();
+  const { enqueueSnackbar } = useSnackbar();
+
+  const [modalStyle] = React.useState(getModalStyle);
+  const [openModal, setOpen] = useState(false);
+  const [courses, setCourses] = useState([]);
+  const [groupName, setGroupName] = useState('');
+  const [courseId, setCourseId] = useState('');
+  const [userProfile, setProfile] = useState([])
+  const [groupError, setGroupErrors] = useState('');
+  const [groupNames, setGroupNames] = useState([]);
+
+  const [uploading, setUploading] = useState(false);
+  const [groupPicture, setGroupPicture] = useState('');
+  const [formValid, setFormValid] = useState(false)
+
+  const onDrop = useCallback(async (droppedFiles) => {
+    if (droppedFiles.length) {
+        setUploading(true);
+
+        const form = new FormData();
+        form.append('image', droppedFiles[0]);
+        const res = await axios.post('/upload/single', form)
+          .catch((err) => console.log(err));
+        if (res && res.data) {
+          setGroupPicture(res.data.imageUrl)
+          setUploading(false)
+        }
+    }
+  }, []);
+
+  const {
+    getRootProps,
+    isDragActive,
+    isDragAccept,
+    isDragReject,
+  } = useDropzone({onDrop, maxFiles: 1, accept: 'image/*'});
+
+  const style = useMemo(() => ({
+    ...baseStyle,
+    ...(isDragActive ? activeStyle : {}),
+    ...(isDragAccept ? acceptStyle : {}),
+    ...(isDragReject ? rejectStyle : {})
+  }), [
+    isDragActive,
+    isDragReject,
+    isDragAccept
+  ]);
+
+  const handleOpen = () => {
+    setOpen(true);
+    setGroupPicture('');
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+    setGroupPicture('');
+    setGroupName('');
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    
+    if(formValid) {
+
+      const data = { groupName: groupName, imageUrl: groupPicture , courseId: courseId};
+      const response = await axios.post('/user/groups', data);
+
```

Comment:
To get this to automatically update the Groups page, modify the backend POST route to return the new group, append it to the users groups you're getting from context and then call the action` fetchUserGroups(userGroups)(dispatch)`.

Reply:
@timegambit this would be a nice approach but I'm not seeing fetchUserGroups action in this branch?

Reply:
@amha19 you're right it's on dev! I should have mentioned @ethanrosenberg and I were considering merging dev into this branch first to make some changes based on what's in dev now.